### PR TITLE
Fix createStd(Err|Out)DiagnosticForSimulator

### PR DESCRIPTION
**Pull request by plu**

### CODE REVIEW

**Comment by plu** [comment]
This is still work in progress, so far there's only a failing test case.

**Comment by plu** [comment]
```diff

diff --git a/FBControlCore/Diagnostics/FBDiagnostic.m b/FBControlCore/Diagnostics/FBDiagnostic.m
index 2878a42..5268025 100644
--- a/FBControlCore/Diagnostics/FBDiagnostic.m
+++ b/FBControlCore/Diagnostics/FBDiagnostic.m
@@ -967,9 +967,6 @@ - (instancetype)updateString:(NSString *)string
 - (instancetype)updatePath:(NSString *)path
 {
   [self flushBackingStore];
-  if (![NSFileManager.defaultManager fileExistsAtPath:path]) {
-    return self;
-  }
   object_setClass(self.diagnostic, FBDiagnostic_Path.class);
   self.diagnostic.backingFilePath = path;
   if (!self.diagnostic.shortName) {
```

@lawrencelomax This makes the test added in 2629ac3 pass, but probably has other side effects.

**Comment by lawrencelomax** [comment]
Thanks!

**Comment by plu** [comment]
Status update:

The two tests I added (1a8e0c0, 8c74479) are passing now. I've added this hack (8c96841) to make `SimDevice` put the file in the desired place.

However I'm unsure about the changes I did here: 03fa1d7

File: FBSimulatorControl/Strategies/FBApplicationLaunchStrategy.m [hunk 1]
```diff
@@ -282,8 +282,26 @@ @implementation FBApplicationLaunchStrategy_CoreSimulator
...
+    stdErrPath:[self translateAbsolutePath:stdErrPath toPathRelativeTo:simulator.dataDirectory]];
   return [simulator.device launchApplicationWithID:appLaunch.bundleID options:options error:error];
 }
 
+- (NSString *)translateAbsolutePath:(NSString *)absolutePath toPathRelativeTo:(NSString *)referencePath
+{
+  if (![absolutePath hasPrefix:@"/"]) {
+    return absolutePath;
+  }
+  // When launching an application with a custom stdout/stderr path, `SimDevice` uses the given path relative
```
Comment (lawrencelomax) [inline FBSimulatorControl/Strategies/FBApplicationLaunchStrategy.m]: Good description, helpful for people in future.

**Pull request closed without merging**